packages/mediapipe-core/lib/src/ffi_utils.dart

```diff
@@ -76,3 +76,21 @@ int _length(Pointer<Uint8> codeUnits) {
   }
   return length;
 }
+
+/// Offers convenience and readability extensions for detecting null pointers.
+extension NullAwarePtr on Pointer {
+  /// Returns true if this is a null pointer.
+  bool get isNullPointer => address == 0;
+
+  /// Returns true if this is not a null pointer.
+  bool get isNotNullPointer => address != 0;
+}
+
+/// Returns true if this nullable pointer is both not-null and not a `NullPtr`,
+/// which is to say, its address is 0x00000000.
+bool isNotNullOrNullPointer(Pointer? ptr) =>
+    ptr != null && ptr.isNotNullPointer;
+
+/// Returns true if this nullable pointer is either null OR is a `NullPtr`,
+/// which is to say, its address is 0x00000000.
+bool isNullOrNullPointer(Pointer? ptr) => ptr == null || ptr.isNullPointer;
```

Comment:
These could actually just be more getters in the extension. :)

Reply:
The idea of having these as helper methods is to avoid the null check, since `null` doesn't have the getter `isNullPointer`. So I believe we'd need to essentially have the body of this method everywhere we currently call the method, and there isn't a clean extension-getter translation available to us.

Does that seem right?

Reply:
Oh right, yup! I see what you mean. SGTM.